### PR TITLE
[docs] Make cell editable in demos

### LINKED ISSUE #1625: [docs] Improve commodity to add edition on columns
How about we do desk as not editable, and all the rest editable? 

Filled quantity can be sort of slider...
Currency can be a dropdown same for status
IsFilled boolean column type

### PR DESCRIPTION
**Pull request by m4theushw**
Closes #1625
To be merged after #1772 
Previews: 
- https://deploy-preview-1817--material-ui-x.netlify.app/components/data-grid/ 
- https://deploy-preview-1817--material-ui-x.netlify.app/components/data-grid/demo/

### CODE REVIEW

File: packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts [hunk 1]
```diff
@@ -147,31 +147,36 @@ export const useGridKeyboard = (
...
+      // See https://github.com/mui-org/material-ui/issues/10534
+      if (event.target instanceof HTMLElement && !isGridCellRoot(event.target)) {
         return;
       }
       if (event.isPropagationStopped()) {
         return;
       }
-      const isEditMode = params.cellMode === 'edit';
+
+      // Get the latest params because the cell mode may have changed by another listener
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts:159]: Fixes a bug I found when pressing the spacebar in a editable cell in view mode.

ivnNjiCVbw

File: docs/src/pages/components/data-grid/overview/DataGridDemo.js [hunk 2]
```diff
@@ -3,13 +3,27 @@ import { DataGrid } from '@material-ui/data-grid';
...
 const columns = [
   { field: 'id', headerName: 'ID', width: 90 },
-  { field: 'firstName', headerName: 'First name', width: 150 },
-  { field: 'lastName', headerName: 'Last name', width: 150 },
+  {
+    field: 'firstName',
+    headerName: 'First name',
+    width: 150,
+    editable: true,
+    disableClickEventBubbling: true,
```
Comment (dtassone) [inline docs/src/pages/components/data-grid/overview/DataGridDemo.js:11]: why do we need this prop?

File: packages/grid/x-grid-data-generator/src/renderer/renderEditProgress.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    debouncedUpdateCellEditProps(newValue);
+  };
+
+  React.useEffect(() => {
+    setValueState(Number(value));
+  }, [value]);
+
+  return (
+    <div className={classes.root}>
+      <Slider
```
Comment (dtassone) [inline packages/grid/x-grid-data-generator/src/renderer/renderEditProgress.tsx:64]: Maybe we can reuse the same style as the view? So we keep it consistent.

Comment (m4theushw) [reply packages/grid/x-grid-data-generator/src/renderer/renderEditProgress.tsx]: Done.

File: docs/src/pages/components/data-grid/overview/DataGridDemo.js (hunk 2 above)
Comment (m4theushw) [reply docs/src/pages/components/data-grid/overview/DataGridDemo.js:11]: It stops the propagation of any click inside the cell. Without it, if you click the cell the row is selected.

HnuGKhIGT4

File: packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts [hunk 4]
```diff
@@ -201,10 +223,11 @@ export function useGridEditRows(apiRef: GridApiRef) {
   );
 
   const handleExitEdit = React.useCallback(
-    (params: GridCellParams, event: React.SyntheticEvent) => {
```
Comment (DanailH) [inline packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts:204]: Are the `params` no longer of the given type?

File: docs/src/pages/components/data-grid/overview/DataGridDemo.js (hunk 2 above)
Comment (dtassone) [reply docs/src/pages/components/data-grid/overview/DataGridDemo.js:11]: we should use disableSelectionOnClick

File: packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts (hunk 4 above)
Comment (m4theushw) [reply packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts:204]: Fixed.

File: docs/src/pages/components/data-grid/overview/DataGridDemo.js (hunk 2 above)
Comment (m4theushw) [reply docs/src/pages/components/data-grid/overview/DataGridDemo.js:11]: Updated!

File: docs/src/pages/components/data-grid/overview/DataGridDemo.tsx [hunk 5]
```diff
@@ -3,13 +3,27 @@ import { DataGrid, GridColDef, GridValueGetterParams } from '@material-ui/data-g
...
 const columns: GridColDef[] = [
   { field: 'id', headerName: 'ID', width: 90 },
-  { field: 'firstName', headerName: 'First name', width: 150 },
-  { field: 'lastName', headerName: 'Last name', width: 150 },
+  {
+    field: 'firstName',
+    headerName: 'First name',
+    width: 150,
+    editable: true,
+    disableClickEventBubbling: true,
```
Comment (dtassone) [inline docs/src/pages/components/data-grid/overview/DataGridDemo.tsx:11]: don't we need `disableSelectionOnClick` here as well?

File: docs/src/pages/components/data-grid/overview/DataGridDemo.js (hunk 2 above)
Comment (dtassone) [reply docs/src/pages/components/data-grid/overview/DataGridDemo.js:11]: FYI I believe we should also be able to handle that with `event.stopPropagation()` directly

File: docs/src/pages/components/data-grid/overview/DataGridDemo.tsx (hunk 5 above)
Comment (m4theushw) [reply docs/src/pages/components/data-grid/overview/DataGridDemo.tsx]: I missed this one.

**Review by dtassone — APPROVED** [review]

File: packages/grid/_modules_/grid/components/GridRow.tsx [hunk 6]
```diff
@@ -33,7 +33,12 @@ export const GridRow = (props: GridRowProps) => {
   const publish = React.useCallback(
     (eventName: string) => (event: React.MouseEvent) => {
       // Ignore portal
-      if (!event.currentTarget.contains(event.target as HTMLElement)) {
+      // The target is not an element when triggered by a Select inside the cell
+      // See https://github.com/mui-org/material-ui/issues/10534
+      if (
+        (event.target as any).nodeType === 1 &&
+        !event.currentTarget.contains(event.target as Element)
```
Comment (dtassone) [inline packages/grid/_modules_/grid/components/GridRow.tsx:40]: BTW I thought I put a comment about this one, or is it on another PR and we are repeating it again?

Comment (m4theushw) [reply packages/grid/_modules_/grid/components/GridRow.tsx:40]: We're no repeating again. This PR is stacked on top of #1772, which is where you added your comment. #1772 has to be reviewed first.

**Comment by m4theushw** [comment]
I've updated this PR to also make the cells editable in the employee dataset.

**Review by dtassone — APPROVED** [review]

**Pull request merged**